drivers/i2c/i2c_gd32.c

```diff
@@ -0,0 +1,684 @@
+/*
+ * Copyright (c) 2021 BrainCo Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT gd_gd32_i2c
+
+#include <errno.h>
+#include <kernel.h>
+#include <devicetree.h>
+#include <drivers/pinctrl.h>
+#include <drivers/i2c.h>
+#include <soc.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(i2c_gd32, CONFIG_I2C_LOG_LEVEL);
+
+#include "i2c-priv.h"
+
+struct i2c_gd32_config {
+	uint32_t reg;
+	uint32_t bitrate;
+	uint32_t rcu_periph_clock;
+	const struct pinctrl_dev_config *pcfg;
+	void (*irq_cfg_func)(void);
+};
+
+struct i2c_gd32_data {
+	struct k_sem bus_mutex;
+	struct k_sem sync_sem;
+	uint32_t dev_config;
+	uint16_t addr1;
+	uint16_t addr2;
+	uint32_t xfer_len;
+	struct i2c_msg *current;
+	uint8_t is_berr;
+	uint8_t is_lostarb;
+	uint8_t is_aerr;
+	uint8_t is_restart;
+	uint8_t is_stopped;
+};
+
+static inline void i2c_gd32_enable(const struct i2c_gd32_config *cfg)
+{
+	I2C_CTL0(cfg->reg) |= I2C_CTL0_I2CEN;
+}
+
+static inline void i2c_gd32_disable(const struct i2c_gd32_config *cfg)
+{
+	I2C_CTL0(cfg->reg) &= ~I2C_CTL0_I2CEN;
```

Comment:
note: some operations, e.g.
```c
I2C_CTL0(cfg->reg) |= I2C_CTL0_STOP;
```
don't have a helper, so why do some others have it?

Reply:
Remove other helper, add comment on it.